atomic_reactor/util.py

```diff
@@ -1567,3 +1567,95 @@ def dump_stacktraces(sig, frame):
 
 def setup_introspection_signal_handler():
     signal.signal(signal.SIGUSR1, dump_stacktraces)
+
+
+class DigestCollector(object):
+
+    def __init__(self, images_digests=None):
+        self._images_digests = images_digests or {}
+
+    def __repr__(self):
+        return "DigestCollector(images_digests={!r})".format(self._images_digests)
+
+    def _key(self, image):
+        """
+        :param ImageName image:
+        :rtype: str
+        :return: image name in format name:tag
+        """
+        return image.to_str(tag=True)
+
+    def __contains__(self, image):
+        """Check if collector contains digests of the specified image
+
+        :param ImageName image: image
+        :rtype: bool
+        :return: True if collector contains digests of the specified image
+        """
+        image_name_tag = self._key(image)
+        return image_name_tag in self._images_digests
+
+    def update_image_digest(self, image, platform, digest):
+        """Update parent image digest for specific platform
+
+        :param ImageName image: image
+        :param str platform: name of the platform/arch (x86_64, ppc64le, ...)
+        :param str digest: digest of the specified image (sha256:...)
+        """
+        image_name_tag = self._key(image)
+
+        image_name = image.to_str(tag=False)
+        name_digest = '{}@{}'.format(image_name, digest)
+
+        image_digests = self._images_digests.setdefault(image_name_tag, {})
+        image_digests[platform] = name_digest
+
+    def get_image_digests(self, image):
+        """Get digest of specified image and platform
```

Comment:
This doesn't seem to describe the method.

Reply:
fixed